src/components/PageWithSideBar/PageWithSideBar.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  Drawer,
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerPanelBody,
+  DrawerPanelContent,
+  Text,
+  TextContent,
+  TextVariants,
+} from '@patternfly/react-core';
+import { PropsWithChildren, useEffect, useState } from 'react';
+
+import { useWindowSizeObserver } from 'hooks/useWindowSizeObserver';
+
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { Toolbar } from 'components/Toolbar/Toolbar';
+import { ToolbarItem } from 'components/Toolbar/ToolbarItem';
+
+const WINDOW_WIDTH_THRESHOLD = 1000;
+
+interface IPageWithSideBarProps {
+  sideBarTitle: string;
+  sideBarContent: React.ReactNode;
+  isExpanded: boolean;
+  onExpand: (isExpanded: boolean) => void;
+}
+
+/**
+ * Component rendering page with a side bar.
+ *
+ * @param sideBarTitle - Title of the side bar
+ * @param sideBarContent - Content of the side bar
+ * @param isExpanded - Whether the side bar is expanded
+ * @param onExpand - Function to toggle expand state of the side bar
+ * @param children - Page content
+ */
+export const PageWithSideBar = ({
+  sideBarTitle,
+  sideBarContent,
+  isExpanded,
+  onExpand,
+  children,
+}: PropsWithChildren<IPageWithSideBarProps>) => {
+  const { windowWidth } = useWindowSizeObserver();
+  const [conditionMet, setConditionMet] = useState<boolean>(false);
```

Comment:
Is not this the same is `isExpanded`?

Reply:
Definitely not, `isExpanded` is also updated by user action.